### ISSUE TITLE
[ALM Accelerator - BUG]: Manual entry not saved for connection references

### ISSUE THREAD

**Issue by BeekvanV** [microsoft/coe-starter-kit#8829]
### Describe the issue

This might be by design, but my expectations were different.

I am working on a project on which User A does not have permissions on Downstream Environment T. 
He does however have Basic User permissions as to fill in the deployment configuration on Environment T. 
User B has prepared connections on Environment T and has given the connection Ids to User A to fill in in the deployment configuration.
However when User A enters the deployment configuration for Environment T and Saves. 
The Manual entry for Connection to Outlook has not been saved. 
On selecting deployment configuration in ALM Accelerator once again. The value for the Outlook connector has emptied and shows the dropdown, which is empty because User A does not have eligible connections on Environment T.
The interesting thing is. When a connector shows "Unknown" instead of "Not connected" User A was able to change the already existing value. 
![image](https://github.com/user-attachments/assets/e4e72481-c3aa-4726-a3a9-df173feb45da)

Things I have checked:
* Looked in the settings in Admin MDA, the outlook connection was not added at all. Not the key, not the value
* Removed the deployment settings to force re-create, no dice.
* Used another user account with more permissions, same behavior
* Looked through the Canvas App code. I found a piece of code on the "Save and Close" button which seems to check if the **dropdown** value has been filled out. In case of manual entry, it has not. Could this be the reason why the connector does not save?
![image](https://github.com/user-attachments/assets/31dbba47-174f-4c18-978c-61c79ccb0b8b)

### Expected Behavior

On adding manual entries to the Connection References for them to save to the User Settings Table on selecting Save and Close.

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

1.0.20240305.1

### Steps To Reproduce
See video:
https://github.com/user-attachments/assets/335e7bc4-7eef-4e00-a2ee-0337d007ac10

1. Configure deployment settings for project on which connection references have not been set.
2. Switch one of them to manual entry.
3. Change value and save and close
4. re-open deployment settings
5. manual entry was not saved.

**Comment by Srikanth-Singapuram** [comment]
@BeekvanV We encountered a similar issue and, as a temporary solution, we shared the connectors to the service account that is capable of managing deployments within the ALM Accelerator environment. This will enable see the connection ID in the dropdown of the ALM Accelerator application.

**Comment by BeekvanV** [comment]
@Srikanth-Singapuram Thanks for the reply. As far as I know, connections cannot be shared with service accounts or user accounts. Only with Service principals.

**Comment by Srikanth-Singapuram** [comment]
@BeekvanV Yes, that's correct we cannot share the standard (OOTB) connectors with a Service account or user. My focus is on custom connectors and the process of creating these standard connectors within the target environment using a Service account, as well as the sharing of custom connectors with the Service account from that environment.

**Comment by Ayesha-320** [comment]
Hi team,
According to https://learn.microsoft.com/en-us/power-platform/alm/pipelines, Power Platform Pipelines are now the recommended approach for Application Lifecycle Management (ALM). The ALM Accelerator is no longer actively maintained or supported. Therefore, this issue is being closed accordingly.  

Best Regards, 
CoE Starter Kit Team
